flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetFileSystemFormatFactory.java

```diff
@@ -0,0 +1,247 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.formats.parquet;
+
+import org.apache.flink.api.common.io.FileInputFormat;
+import org.apache.flink.api.common.io.InputFormat;
+import org.apache.flink.api.common.serialization.BulkWriter;
+import org.apache.flink.api.common.serialization.Encoder;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.core.fs.FileInputSplit;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.formats.parquet.row.ParquetRowDataBuilder;
+import org.apache.flink.formats.parquet.utils.SerializableConfiguration;
+import org.apache.flink.formats.parquet.vector.ParquetColumnarRowSplitReader;
+import org.apache.flink.formats.parquet.vector.ParquetSplitReaderUtil;
+import org.apache.flink.table.dataformat.BaseRow;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.filesystem.FileSystemFormatFactory;
+import org.apache.flink.table.filesystem.PartitionPathUtils;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.parquet.hadoop.ParquetOutputFormat;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+import static org.apache.flink.table.dataformat.vector.VectorizedColumnBatch.DEFAULT_SIZE;
+import static org.apache.flink.table.descriptors.FormatDescriptorValidator.FORMAT;
+import static org.apache.flink.table.filesystem.RowPartitionComputer.restorePartValueFromType;
+
+/**
+ * Parquet {@link FileSystemFormatFactory} for file system.
+ */
+public class ParquetFileSystemFormatFactory implements FileSystemFormatFactory {
+
+	public static final ConfigOption<Boolean> UTC_TIMEZONE = key("format.utc-timezone")
+			.booleanType()
+			.defaultValue(false)
+			.withDescription("Use UTC timezone or local timezone to the conversion between epoch" +
+					" time and LocalDateTime. Hive 0.x/1.x/2.x use local timezone. But Hive 3.x" +
+					" use UTC timezone");
+
+	/**
+	 * Parquet properties, start with "parquet", see more in {@link ParquetOutputFormat}.
+	 * - parquet.compression
+	 * - parquet.block.size
+	 * - parquet.page.size
+	 * - parquet.dictionary.page.size
+	 * - parquet.writer.max-padding
+	 * - parquet.enable.dictionary
+	 * - parquet.validation
+	 * - parquet.writer.version
+	 * ...
+	 */
+	public static final String PARQUET_PROPERTIES = "format.parquet";
```

Comment:
use PARQUET_PROPERTIES_KEY here？

Reply:
Just like `FormatDescriptorValidator.FORMAT`, I will add comments to say this is for prefix.